### DISCUSSION TITLE
What front matter options does confluence-html support?

### DISCUSSION THREAD

**Discussion by baslat** [quarto-dev/quarto-cli#7193] [Q&A]
### Description

What front matter options does confluence-html support? I tried using some of the options for html, but they don't seem to have any effect (see below for yaml). I'm particularly interested in changing the figure size and format, thanks!

````qmd
---
title: "My doc"
date: last-modified
author: me
format:
    confluence-html:
        toc: true
        number-sections: true
        fig-width: 12
        fig-format: svg
knitr:
    opts_chunk:
      collapse: true
      comment: "#>"
      echo: false
      message: false
      warning: false
---

````

**Comment by dragonstyle** [comment]
Those options should be supported (e.g. manipulating the figure size really configures the knitr engine in this case, which should be supported across formats). What is making you think that isn't working?

(I noticed that your example begins with only 2 `-` characters, making it an invalid YAML block- just pointing out in case this is the issue)

**Reply by baslat** [reply]
Thanks. The `--` was just a typo, not the root cause. I am getting inconsistent results, so it's hard to pin down what's going on.

If I set (fig-width under format)
```
format:
    confluence-html:
      toc: true
      number-sections: true
      fig-width: 12
      fig-format: svg
knitr:
    opts_chunk:
      collapse: true
      comment: "#>"
      echo: false
      message: false
      warning: false
```
the rendered images on confluence are not wider, they are they default width.

If I set (fig.width under knitr)
```
format:
    confluence-html:
      toc: true
      number-sections: true
      fig-format: svg
knitr:
    opts_chunk:
      collapse: true
      comment: "#>"
      echo: false
      message: false
      warning: false
      fig.width: 12
```
The last image (of 4) is wider, but none of the others. Looking at the attachments page on Confluence I can see that only one image has been newly uploaded, and the other 3 are still from the previous render.

And then if I change it to (fig width wider)
```
format:
    confluence-html:
      toc: true
      number-sections: true
      fig-format: svg
knitr:
    opts_chunk:
      collapse: true
      comment: "#>"
      echo: false
      message: false
      warning: false
      fig.width: 13
```

Nothing changes when published, and it's still 3 at default, 1 at 12.

All images are ggplots.

**Reply by mcanouil** [reply]
Could you share something more reproducible (and small)?

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````


Also what's the output of `quarto check`?

**Reply by baslat** [reply]
fig-width does not change the output.

````qmd
---
title: "Reproducible Quarto Document"
format:
  confluence-html:
    toc: true
    number-sections: true
    fig-width: 12
---

This is a reproducible Quarto document using `format: confluence-html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
library(ggplot2)
ggplot(mtcars, aes(x = wt, y = mpg)) + geom_point()
```

The end
````

`quarto check` results:

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.8.18
      Path: /Users/bas/.pyenv/versions/3.8.18/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Users/bas/repos/asgs/renv/library/R-4.3/aarch64-apple-darwin20
        - /Users/bas/Library/Caches/org.R-project.R/R/renv/sandbox/R-4.3/aarch64-apple-darwin20/ac5c2659
      knitr: 1.44
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK

```

**Reply by baslat** [reply]
It seems as though updated images are not being uploaded to confluence. Through trial and error I am now thinking that `fig.width` under `confluence-html` has no effect, and `fig-width` under `knitr` does have an effect, however, when the rendered document is being uploaded to confluence, not all the new images get pushed. I wrote a hacky script that deletes all attachments from the confluence page before publishing, and now I can reliably change the size of all images with the `knitr` option (but having hacky scripts obviously isn't the best).

And there is no table of contents nor numbered sections either, so I'm still not sure what options confluence-html supports.

````qmd
---
format:
    confluence-html:
      toc: true
      number-sections: true
      self-contained: true
    # this does not affect the outcome
      fig.width: 12
knitr:
    opts_chunk:
      collapse: true
      comment: "#>"
      echo: false
      message: false
      warning: false
      # these do work, but the new images don't always get uploaded to confluence
      fig.width: 12
      fig.height: 7.5
---

````

**Reply by mcanouil** [reply]
Correct options in Quarto uses `-` not `.` and since 1.44 `knitr` also supports all options using `-`.
